### ISSUE TITLE
"The BC provider no longer provides an implementation for KeyFactory.RSA."

### ISSUE THREAD

**Issue by sal0max** [Hakky54/sslcontext-kickstart#139]
**Describe the bug**
On Android >= Android 9/P (API 28) I get the following error in logcat:
```
FATAL EXCEPTION: DefaultDispatcher-worker-1
    Process: myapp, PID: 28762
    nl.altindag.ssl.exception.PrivateKeyParseException: org.bouncycastle.openssl.PEMException: unable to convert key pair: The BC provider no longer provides an implementation for KeyFactory.RSA.  Please see https://android-developers.googleblog.com/2018/03/cryptography-changes-in-android-p.html for more details.
        at nl.altindag.ssl.util.PemUtils.extractPrivateKey(PemUtils.java:454)
        at nl.altindag.ssl.util.PemUtils.$r8$lambda$s6ElOTWwPljNhjlxXLkUrjqZZdY(Unknown Source:0)
        at nl.altindag.ssl.util.PemUtils$$ExternalSyntheticLambda10.apply(Unknown Source:2)
        at java.util.Optional.map(Optional.java:211)
        at nl.altindag.ssl.util.PemUtils.parsePrivateKey(PemUtils.java:421)
        at nl.altindag.ssl.util.PemUtils.parseIdentityMaterial(PemUtils.java:334)
        at nl.altindag.ssl.util.PemUtils.loadIdentityMaterial(PemUtils.java:306)
        at nl.altindag.ssl.util.PemUtils.loadIdentityMaterial(PemUtils.java:232)
        at nl.altindag.ssl.util.PemUtils.loadIdentityMaterial(PemUtils.java:224)
        ...
     Caused by: org.bouncycastle.openssl.PEMException: unable to convert key pair: The BC provider no longer provides an implementation for KeyFactory.RSA.  Please see https://android-developers.googleblog.com/2018/03/cryptography-changes-in-android-p.html for more details.
        at org.bouncycastle.openssl.jcajce.JcaPEMKeyConverter.getPrivateKey(Unknown Source:48)
        at nl.altindag.ssl.util.PemUtils.extractPrivateKey(PemUtils.java:452)
        at nl.altindag.ssl.util.PemUtils.$r8$lambda$s6ElOTWwPljNhjlxXLkUrjqZZdY(Unknown Source:0) 
        at nl.altindag.ssl.util.PemUtils$$ExternalSyntheticLambda10.apply(Unknown Source:2) 
        at java.util.Optional.map(Optional.java:211) 
        at nl.altindag.ssl.util.PemUtils.parsePrivateKey(PemUtils.java:421) 
        at nl.altindag.ssl.util.PemUtils.parseIdentityMaterial(PemUtils.java:334) 
        at nl.altindag.ssl.util.PemUtils.loadIdentityMaterial(PemUtils.java:306) 
        at nl.altindag.ssl.util.PemUtils.loadIdentityMaterial(PemUtils.java:232) 
        at nl.altindag.ssl.util.PemUtils.loadIdentityMaterial(PemUtils.java:224) 
        ...
     Caused by: java.security.NoSuchAlgorithmException: The BC provider no longer provides an implementation for KeyFactory.RSA.  Please see https://android-developers.googleblog.com/2018/03/cryptography-changes-in-android-p.html for more details.
        at sun.security.jca.Providers.checkBouncyCastleDeprecation(Providers.java:386)
        at sun.security.jca.Providers.checkBouncyCastleDeprecation(Providers.java:336)
        at java.security.KeyFactory.getInstance(KeyFactory.java:235)
        at org.bouncycastle.jcajce.util.NamedJcaJceHelper.createKeyFactory(Unknown Source:2)
        at org.bouncycastle.openssl.jcajce.JcaPEMKeyConverter.getKeyFactory(Unknown Source:20)
        at org.bouncycastle.openssl.jcajce.JcaPEMKeyConverter.getPrivateKey(Unknown Source:4)
```

The error links to this site: https://android-developers.googleblog.com/2018/03/cryptography-changes-in-android-p.html

This is my code, how I use this library:

```kotlin
// load private client cert
val cert = javaClass.getResourceAsStream("/credentials/myCert.pem")
val key = javaClass.getResourceAsStream("/credentials/myKey.pem")
if (cert == null || key == null)
    throw Exception("Private certificate or key missing.")
val keyManager = PemUtils.loadIdentityMaterial(cert, key)

// activate client ssl in fuel
val sslFactory = SSLFactory.builder()
    .withIdentityMaterial(keyManager)
    .build()
FuelManager.instance.socketFactory = sslFactory.sslSocketFactory
```

**Environmental Data:**
 - Android >= Android 9/P (API 28)
 - io.github.hakky54:sslcontext-kickstart-for-pem:7.2.0
 - I tested several emulators with different Android versions, and it works on devices < API 28, and doesn't work on devices >= API 28

**Comment by Hakky54** [comment]
Hi @sal0max 

Thank you for submitting this issue. This is indeed a bug on this library as Android does not permit to use BC as a direct provider from version 28 onwards when using some methods of BounceCastle. I validated on my side and I also get the same exception. I have made a fix which I also will be releasing whenever the pull request builds successfully.

**Comment by Hakky54** [comment]
Hi @sal0max I just released this patch on version 7.2.1 Can you retry and share your results?

**Comment by sal0max** [comment]
Great job! Can confirm: Works like a charm.
Thanks so much for the blazing fast fix and in general for this extremely useful library.

Outstanding work. 👍
